evals/evaluation/autorag/evaluation/ragas_evaluation_benchmark.py

```diff
@@ -0,0 +1,200 @@
+        post_json_data["top_n"] = top_n
+        post_json_data["temperature"] = temperature
+        post_json_data["top_k"] = top_k
+        post_json_data["top_p"] = top_p
+        post_json_data["repetition_penalty"] = repetition_penalty
+        json_data = json.dumps(post_json_data)
+
+        headers = {"Content-Type": "application/json"}
+        response = requests.post(url, data=json_data, headers=headers)
+        rag_answer = response.text
+
+        data = {
+            "question": question,
+            "answer": rag_answer,
+        }
+        with jsonlines.open(result_answer_path, "a") as file_json:
+            file_json.write(data)
+
+    contexts_list = load_set(ground_truth_file, "context")
+    ground_truth_list = load_set(ground_truth_file, "ground_truth")
+    question_list = load_set(result_answer_path, "question")
+    answer_list = load_set(result_answer_path, "answer")
+
+    data_samples = {
+        "question": question_list,
+        "answer": answer_list,
+        "contexts": contexts_list,
+        "ground_truth": ground_truth_list,
+    }
+
+    dataset = Dataset.from_dict(data_samples)
+
+    if use_openai_key:
+        os.environ["OPENAI_API_KEY"] = os.getenv("OPENAI_API_KEY")
+        score = evaluate(dataset, metrics=[answer_relevancy, faithfulness, context_recall, context_precision])
+    else:
+        try:
+            if isinstance(llm, str):
+                langchain_llm = HuggingFacePipeline.from_model_id(
+                    model_id=llm,
+                    task="text-generation",
+                    pipeline_kwargs={"max_new_tokens": 128},
+                )
+            else:
+                langchain_llm = llm
+            langchain_llm = LangchainLLMWrapper(langchain_llm)
+
+        except:
+            print("Please check the selected llm!")
+
+        langchain_embeddings = HuggingFaceBgeEmbeddings(
+            model_name=embedding_model,
+            encode_kwargs={"normalize_embeddings": True},
+            query_instruction="Represent this sentence for searching relevant passages:",
+        )
+
+        langchain_embedding = LangchainEmbeddingsWrapper(langchain_embeddings)
+        ### Note: due to the code error in RAGAS repo, do not recommend other model except chatgpt to
+        # evaluate context_recall and context_precision.
+        # Please refer https://github.com/explodinggradients/ragas/issues/664.
+        score = evaluate(
+            dataset,  # pylint: disable=E1123
+            metrics=[answer_relevancy, faithfulness],
+            llm=langchain_llm,  # pylint: disable=E1123
+            embeddings=langchain_embedding,
+        )  # pylint: disable=E1123
+
+    df = score.to_pandas()
+    answer_relevancy_average = df["answer_relevancy"][:].mean()
+    faithfulness_average = df["faithfulness"][:].mean()
+    context_recall_average = df["context_recall"][:].mean()
+    context_precision_average = df["context_precision"][:].mean()
```

Comment:
the `context_recall` and `context_precision` is not evaluated at line 121, so should remove the 2 metrics?

Reply:
Are there still issues with the 2 metrics at ragas? refer [this pr](https://github.com/opea-project/GenAIEval/pull/37)

Reply:
use openai api could support the other 2 metrics, will add the output criteria